CONTRIBUTING.md

```diff
@@ -0,0 +1,65 @@
+
+To install the CLI locally, run the following commands from the root project directory:
+```
+poetry lock      # only needed if you updated dependencies in pyproject.toml
+poetry install
+```
+You do not need to re-run these commands each time you update code locally, unless you've added dependencies in pyproject.toml.
+
+
+## Developing
+If you do update dependencies in `pyproject.toml`, run `poetry lock` and check in the resulting changes to `poetry.lock` along with the rest of 
+your code changes.
+
+### Tests, linters, and formatting
+To run the tests, execute the following command from the root project (terra-scientific-pipelines-service-cli) directory:
+```bash
+poetry run pytest
+```
+
+To run tests with a coverage report printed to the terminal:
+```bash
+poetry run pytest --cov-report term --cov=terralab
+```
+
+To run the formatter, execute the following command from the root project directory:
+```bash
+poetry run black .
+```
+
+To run the linter with fixes, execute the following command from the root project directory:
+```bash
+poetry run ruff check --fix
+```
+To run the linter as a check without fixes, omit the `--fix` flag.
+
+## Coordinating changes with the Teaspoons service
+Sometimes, updates to [Teaspoons](https://github.com/DataBiosphere/terra-scientific-pipelines-service) will need to be coordinated with updates to the CLI. 
+
+To test changes in Teaspoons locally against your local CLI, do the following once you've made changes in your local code and are ready to test them:
+
+1. In Teaspoons, run `./gradlew openApiGenerate` to create the autogenerated Python client locally (the files will be generated in `teaspoons-cli/generated` in that repo). You do not need to move these files (and they will not be checked in to the Teaspoons repo).
+2. In this repo, edit the [pyproject.toml](pyproject.toml) file as follows: comment out the line that defines the `terra-scientific-pipelines-service-api-client` dependency, and add a line like `teaspoons-client = {path = "YOUR/LOCAL/PATH/TO/terra-scientific-pipelines-service/python-client/generated"}`
+ It should look something like this:
+ ```
+ teaspoons-client = {path = "/Users/awesomedeveloper/workbench/terra-scientific-pipelines-service/python-client/generated"}
+# terra-scientific-pipelines-service-api-client = "^0.1.0"
+```
+3. Run `poetry lock` and `poetry install`.
+
+Your CLI should now be importing the `teaspoons_client` via the local code rather than the package from PyPi.
+
+Note: if the locally generated code is not being imported as expected, try deleting and recreating the poetry virtual environment by running the following:
+```
+rm -rf $(poetry env info --path)
+poetry install
+```
+
+To run the CLI against your locally running instance of Teaspoons:
+4. In Teaspoons, run `./gradlew bootRun` and wait for the service to spin up locally.
```

Comment:
i think this line can point to the readme at "how to run teaspoons" just in case that changes we dont ahve to update it here too